pkg/configuration/new_asset_store.go

```diff
@@ -4,50 +4,43 @@ import (
+// configuration and CAS
+func NewAssetStoreFromConfiguration(configuration *pb.AssetCacheConfiguration, contentAddressableStorage blobstore_configuration.BlobAccessInfo, grpcClientFactory grpc.ClientFactory, maximumMessageSizeBytes int) (storage.AssetStore, error) {
+	if configuration == nil {
+		return nil, nil
+	}
 	switch backend := configuration.Backend.(type) {
 	case *pb.AssetCacheConfiguration_BlobAccess:
 		assetBlobAccessCreator := asset_configuration.NewAssetBlobAccessCreator(grpcClientFactory, maximumMessageSizeBytes)
 
 		assetBlobAccess, err := blobstore_configuration.NewBlobAccessFromConfiguration(
-			backend.BlobAccess.AssetStore,
+			backend.BlobAccess,
 			assetBlobAccessCreator)
 		if err != nil {
-			return nil, nil, err
-		}
-		assetStore = storage.NewBlobAccessAssetStore(assetBlobAccess.BlobAccess, maximumMessageSizeBytes)
-		contentAddressableStorageInfo, err := blobstore_configuration.NewBlobAccessFromConfiguration(backend.BlobAccess.ContentAddressableStorage, blobstore_configuration.NewCASBlobAccessCreator(grpcClientFactory, maximumMessageSizeBytes))
-		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
-		contentAddressableStorage = contentAddressableStorageInfo.BlobAccess
+		return storage.NewBlobAccessAssetStore(assetBlobAccess.BlobAccess, maximumMessageSizeBytes), nil
 	case *pb.AssetCacheConfiguration_ActionCache:
-		cas, actionCache, err := blobstore_configuration.NewCASAndACBlobAccessFromConfiguration(backend.ActionCache.Blobstore, grpcClientFactory, maximumMessageSizeBytes)
-		if err != nil {
-			return nil, nil, err
-		}
-		contentAddressableStorage = cas
-		assetStore = storage.NewActionCacheAssetStore(actionCache, contentAddressableStorage, maximumMessageSizeBytes)
-	case *pb.AssetCacheConfiguration_CasOnly:
-		contentAddressableStorageInfo, err := blobstore_configuration.NewBlobAccessFromConfiguration(backend.CasOnly.ContentAddressableStorage, blobstore_configuration.NewCASBlobAccessCreator(grpcClientFactory, maximumMessageSizeBytes))
+		actionCache, err := blobstore_configuration.NewBlobAccessFromConfiguration(
+			backend.ActionCache,
+			blobstore_configuration.NewACBlobAccessCreator(
+				contentAddressableStorage,
+				grpcClientFactory,
+				maximumMessageSizeBytes))
 		if err != nil {
-			return nil, nil, err
+			return nil, err
 		}
-		contentAddressableStorage = contentAddressableStorageInfo.BlobAccess
+		return storage.NewActionCacheAssetStore(actionCache.BlobAccess, contentAddressableStorage.BlobAccess, maximumMessageSizeBytes), nil
 	default:
-		return nil, nil, status.Errorf(codes.InvalidArgument, "Asset Cache configuration is invalid as no supported Asset Cache is defined.")
+		return nil, status.Errorf(codes.InvalidArgument, "Asset Cache configuration is invalid as no supported Asset Cache is defined.")
 	}
-	return assetStore, contentAddressableStorage, nil
+	return nil, status.Errorf(codes.Internal, "Something went wrong creating Asset Cache.")
```

Comment:
NIT: Very windows esque error 
![image](https://user-images.githubusercontent.com/7274845/114008819-832a5400-985a-11eb-91d6-73d204304528.png)

This is a code path we never expect to reach, are we able to remove this return statement?